frog/imports/ui/App/index.js

```diff
@@ -34,12 +34,19 @@ export default class App extends Component {
   constructor() {
     super();
     this.state = { app: 'home', username: '' };
-    Meteor.subscribe('userData', { onReady: this.handleNewHash });
+    Meteor.subscribe('userData', {
+      onReady: () => {
+        this.handleNewHash();
+        window.addEventListener('hashchange', this.handleNewHash, false);
+      }
+    });
   }
 
   handleNewHash = () => {
     const [, username, location] = window.location.hash.split('/');
-    if (username) {
+    const loggedInUsername = Meteor.users.findOne(Meteor.userId()).username;
```

Comment:
I don't know if it's relevant since we don't really handle different users and since students can only access students view, but with an URL like this one : http://localhost:3000/student1 the window.location.hash will be "". That means that username and location will be set to undefined